controllers/azurejson_machinepool_controller.go

```diff
@@ -147,6 +142,17 @@ func (r *AzureJSONMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl
 		return reconcile.Result{}, errors.Wrap(err, "failed to create scope")
 	}
 
+	// Construct secret for this machine
```

Comment:
Can we add a TODO comment here about removing this once cloud-provider-azure supports user assigned IDs as resource IDs?

Reply:
same thing in the other controllers